### PR TITLE
 update the client generator to set a client-side timeout

### PR DESCRIPTION
**Pull request by deads2k**
Our client-go/client has a way to set a context deadline based on a timeout.  However, the ListOptions timeoutSeconds didn't trigger this flow since the parameters were built differently.  This updates the generated client to set the local timeout as well.  This helps in cases where the server hangs unexpectedly or the network fails in some way that prevent the server from closing the request.

@kubernetes/sig-api-machinery-bugs 
@sttts @mfojtik @knobunc @ironcladlou @caesarxuchao fyi

```release-note
timeouts set in ListOptions for clients will also be respected locally
```

### CODE REVIEW

File: staging/src/k8s.io/code-generator/_examples/apiserver/clientset/internalversion/typed/example/internalversion/testtype.go [hunk 1]
```diff
@@ -150,10 +162,15 @@ func (c *testTypes) Delete(name string, options *v1.DeleteOptions) error {
 
 // DeleteCollection deletes a collection of objects.
 func (c *testTypes) DeleteCollection(options *v1.DeleteOptions, listOptions v1.ListOptions) error {
+	var timeout time.Duration = 0
+	if opts.TimeoutSeconds != nil {
```
Comment (smarterclayton) [inline staging/src/k8s.io/code-generator/_examples/apiserver/clientset/internalversion/typed/example/internalversion/testtype.go]: SHouldn't this be listOptions?

File: pkg/client/clientset_generated/internalclientset/typed/admissionregistration/internalversion/validatingwebhookconfiguration.go [hunk 2]
```diff
@@ -72,21 +74,31 @@ func (c *validatingWebhookConfigurations) Get(name string, options v1.GetOptions
...
 		VersionedParams(&opts, scheme.ParameterCodec).
+		Timeout(timeout).
 		Do().
 		Into(result)
 	return
 }
 
 // Watch returns a watch.Interface that watches the requested validatingWebhookConfigurations.
 func (c *validatingWebhookConfigurations) Watch(opts v1.ListOptions) (watch.Interface, error) {
+	var timeout time.Duration = 0
```
Comment (smarterclayton) [inline pkg/client/clientset_generated/internalclientset/typed/admissionregistration/internalversion/validatingwebhookconfiguration.go]: `var timeout time.Duration`

would be correct.  Adding zero values is redundant.

**Comment by deads2k** [comment]
nits addressed

**Comment by deads2k** [comment]
rebase and test added.

**Pull request closed without merging**